### ISSUE TITLE
Instruct user to open EX app before starting verification

### ISSUE THREAD

**Issue by mxandreas** [element-hq/element-web#28520]
When verifying the device with another device, the EX app needs to be open before the verification is started, otherwise the request is lost. This is not ideal, but there is no trivial fix for it. Therefore have chosen to instruct the user to open the app.

On EX side there, there is an extra step between selecting another device as means of verification and actually starting the verification. To keep things currently simple on EW, simply update the copy in this modal:

![image](https://github.com/user-attachments/assets/226163f3-adab-4f72-83b2-0b9e18dda254)

"Verify your identity to access encrypted messages and prove your identity to others. If you use another mobile device, please open the app there before you proceed."

This has already created [confusion](https://github.com/element-hq/element-x-android/issues/3872), thus important to bring some clarity ASAP.

**Comment by t3chguy** [comment]
The user could be using any device to verify it, including another session on the same device, so saying "another mobile device" is very confusing

**Comment by mxandreas** [comment]
Yes, they could. That is why it is says that **if** you use a mobile device.

**Comment by t3chguy** [comment]
"another mobile device"? I don't have "another mobile device" I have just the oen

**Comment by t3chguy** [comment]
> When verifying the device with another device, the EX app needs to be open before the verification is started, otherwise the request is lost. This is not ideal, but there is no trivial fix for it.

Where is the issue with context for why this is not fixed?

**Comment by mxandreas** [comment]
> "another mobile device"? I don't have "another mobile device" I have just the one

I do not think many users run EW on their mobile device, if that is what you meant. "Another mobile device" does not in my opinion suggest that you have multiple mobile devices - it is another device that is mobile. However, I do not mind alternative wordings which may be more clear.

**Comment by frebib** [comment]
> the EX app needs to be open before the verification is started, otherwise the request is lost.

Classic Element (Android) doesn't have this issue so it doesn't sound like it's impossible to implement. Is this an iOS specific limitation, or just a Hard Problem™ that isn't going to be attempted to be fixed?

**Comment by stefanceriu** [comment]
The underlying issue here is that in order to [bootstrap the session verification listeners](https://github.com/matrix-org/matrix-rust-sdk/blob/main/bindings/matrix-sdk-ffi/src/client.rs#L780) we need access to the user's identity which is dependant on a keys query response. Normally that would be handled through an olm machine listener but, as I understand it, those get reset often due to missing cross proccess locking. As such, we use sync callbacks as a retry mechanism until we can actually get our hands on a `session_verification_controller`. Obviously, this isn't ideal but the fix isn't at all straight forward.

**Comment by BillCarsonFr** [comment]
> The underlying issue here is that in order to [bootstrap the session verification listeners](https://github.com/matrix-org/matrix-rust-sdk/blob/main/bindings/matrix-sdk-ffi/src/client.rs#L780) we need access to the user's identity which is dependant on a keys query response. Normally that would be handled through an olm machine listener but, as I understand it, those get reset often due to missing cross proccess locking. As such, we use sync callbacks as a retry mechanism until we can actually get our hands on a `session_verification_controller`. Obviously, this isn't ideal but the fix isn't at all straight forward.

I thought the issue was that the notification client was swallowing the to_device verification request events

**Comment by richvdh** [comment]
> "Another mobile device" does not in my opinion suggest that you have multiple mobile devices

I think it does mean exactly that to a native English speaker, I'm afraid. I think you want "If you *also* use a mobile device, ..."

**Comment by andybalaam** [comment]
Updated at https://localazy.com/p/element-web/source-language/_a7632610702617169319 . I was not able to ask for review in localazy, so I asked for review in #element-dev:matrix.org

**Comment by andybalaam** [comment]
Review will happen when the localazy change is downloaded and merged into the code base.

**Comment by andybalaam** [comment]
Done in https://github.com/element-hq/element-web/pull/28573
